docs/src/account/overview.md

```diff
@@ -0,0 +1,61 @@
+# Accounts / Smart Contracts
+
+An `Account` represents the primary entity in Miden. It is capable of holding assets, storing data, and executing custom code. Each `Account` is a smart contract with a programmable interface through which note and transaction scripts can interact with the account's state and assets. By executing [transactions](../transaction.md) against an account, its state can be modified.
+
+## What is the purpose of an account?
+
+In Miden's hybrid UTXO- and account-based model, accounts enable the creation of expressive smart contracts via a Turing-complete language. As examples, an account with a wallet interface can hold the assets of a user while a DEX account could have an interface to trade assets.
+
+## Account composition
+
+An `Account` is composed of several core parts, illustrated below:
+
+<p style="text-align: center;">
+    <img src="../img/account/account-definition.png" style="width:30%;" alt="Account diagram"/>
+</p>
+
+These parts are:
+
+1. [ID](id.md)
+2. [Code](code.md)
+3. [Storage](storage.md)
+4. [Vault](#vault)
+5. [Nonce](#nonce)
+
+### Vault
+
+> [!Note]
+> A collection of [assets](../asset.md) stored by the `Account`.
+
+Large amounts of fungible and non-fungible assets can be stored in the account's vault.
+
+### Nonce
+
+> [!Note]
+> A counter incremented with each state update to the `Account`.
+
+The nonce ensures that an account has a unique _commitment_ (or "hash") after every transaction, even if it contains the same assets and has the same storage state. That in turn allows ordering of transactions and prevents replay attacks. Whenever the state of an account changes in a transaction, its nonce must be incremented and it can only be incremented exactly once per transaction.
+
+Note that a transaction does not always change the state of an account. For instance, a transaction in which two SWAP notes are matched together does not necessarily change anything about the account state. Consequently, the nonce does not have to be incremented.
+
+## Account lifecycle
+
+Throughout its lifetime, an `Account` progresses through various phases:
+
+- **Creation and Deployment:** Initialization of the `Account` on the network.  
+- **Active Operation:** Continuous state updates via `Account` functions that modify the storage, nonce, and vault.
+- **Termination or Deactivation:** Optional, depending on the contract's design and governance model. Note however, that there is no functionality in the protocol to delete an account.
```

Comment:
Since this is not enshrined in the protocol, I wonder if we really need this part?